packages/mosaic-tanstack-table-core/src/trimmed/data-table.ts

```diff
@@ -166,32 +167,74 @@ export class MosaicDataTable<
     }
 
     // Add column filters
+    const internalClauses: Array<mSql.FilterExpr> = [];
+
     columnFilters.forEach((columnFilter) => {
       const columnAccessor = this.#columnDefIdToSqlColumnAccessor.get(
         columnFilter.id,
-      )!; // Assertion is safe due to filtering above
-      // Build the filter expression based on the filter value
-
-      // TODO: Support different filter types based on ColumnDef meta.
-      // TODO: Move this building logic into its own utility function/method.
-      // TODO: Figure out the best usage for AND and OR combinations.
-
-      // POC of what this could look like
-      if (columnFilter.value && typeof columnFilter.value === 'string') {
-        // Simple equals filter for now
-        whereClauses.push(
-          mSql.sql`${columnAccessor} ILIKE ${mSql.literal(columnFilter.value.trim())}`,
-        );
+      )!;
+      // Assertion is safe due to filtering above
+
+      // Find the Column Definition to check for metadata
+      const colDef = this.#store.state.columnDefs.find(
```

Comment:
Is this not accessible using any of Maps defined on the Class? With this implementation, we are having loop-over all the column definitions to find the column. Using a Map would result in constant-time-lookup.